automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Component.scala

```diff
@@ -63,8 +66,37 @@ abstract class Component(queryString: QueryString)(implicit webDriver: WebDriver
 
   override def awaitReady(): Unit = awaitVisible()
 
+  /**
+    * References:
+    *  .getBoundingClient browser compatibility: https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect
+    *  viewport script source: https://stackoverflow.com/questions/123999/how-to-tell-if-a-dom-element-is-visible-in-the-current-viewport
+    *
+    *  When there is a vertical scroll on page and WebElement is outside of view, scroll WebElement into view
+    */
   def scrollToVisible(): Unit = {
-    webDriver.asInstanceOf[JavascriptExecutor].executeScript("arguments[0].scrollIntoView(true)", find(query).get.underlying)
+    // has a vertical scrollbar?
```

Comment:
Execute Javascript `scrollToVisible` on a WebElement before click in `doClick` is needed only on invisible WebElement (outside of viewport). Execute this JS moves WebElement to top of page, led to unpredictable click action because WebDriver is too fast.

Reply:
If we have to scroll, that's fine. And if the built-in `Element.scrollIntoView()` doesn't work, that's unfortunate but also fine(-ish). My point was that there was a time in the past (possibly before you got here) that we didn't have to do this at all; clicking elements outside of the viewport worked. My impression at the time was that it was a bug so I thought scrolling was a temporary measure. Therefore, if I was confronted with having to do this much work, I would have tried just removing the scrolling first. I wanted to make sure you'd tried that.

Now that I look at the code more closely (thank you for extracting the methods... that helps readability), I wonder if you're scrolling conditionally only so you don't have to micro-sleep if no scrolling is necessary. If that's the case, there may be another option. According to https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView, you can ask that the transition be instant:
```
element.scrollIntoView({behavior: "instant"});
```
Have you tried that?

Reply:
I have not tried either before. But I'd like to try removing the call `scrollToVisible` first. I'm little concerned with time it would take to test out this small change. Few passes in Jenkins aren't enough for prove solution works every time. I've been repeatedly Jenkins testing this PR for a week by now and I haven't seen at all any test fails related to click. With that said, change made to use `element.scrollIntoView({behavior: "instant"});`